test/qtism/data/storage/xml/marshalling/ItemResultMarshallerTest.php

```diff
@@ -0,0 +1,166 @@
+        ');
+
+        $this->assertInstanceOf(ItemResult::class, $itemResult);
+
+        $this->assertEquals('fixture-identifier', $itemResult->getIdentifier()->getValue());
+        $this->assertEquals('fixture-identifier', $itemResult->getIdentifier());
+
+        $this->assertInstanceOf(DateTime::class, $itemResult->getDatestamp());
+
+        $this->assertEquals('initial', SessionStatus::getNameByConstant($itemResult->getSessionStatus()));
+
+        $this->assertFalse($itemResult->hasSequenceIndex());
+        $this->assertNull($itemResult->getSequenceIndex());
+        $this->assertFalse($itemResult->hasCandidateComment());
+        $this->assertNull($itemResult->getCandidateComment());
+        $this->assertFalse($itemResult->hasItemVariables());
+        $this->assertNull($itemResult->getItemVariables());
+    }
+
+    public function testMarshall()
+    {
+        $component = new ItemResult(
+            new QtiIdentifier('fixture-identifier'),
+            new DateTime('2018-06-27T09:41:45.529'),
+            1,
+            new ItemVariableCollection(array(
+                new ResultResponseVariable(
+                    new QtiIdentifier('response-identifier'), 0, new CandidateResponse()
+                ),
+                new ResultTemplateVariable(
+                    new QtiIdentifier('response-identifier'), 0
+                )
+            )),
+            new QtiString('candidate-comment'),
+            new QtiInteger(1)
+        );
+
+        /** @var DOMElement $element */
+        $element = $this->getMarshallerFactory()->createMarshaller($component)->marshall($component);
+
+        $this->assertInstanceOf(DOMElement::class, $element);
+
+        $this->assertEquals($component->getQtiClassName(), $element->nodeName);
+
+        $this->assertEquals('fixture-identifier', $element->getAttribute('identifier'));
+        $this->assertTrue($element->hasAttribute('datestamp'));
+        $this->assertEquals('initial', $element->getAttribute('sessionStatus'));
+        $this->assertEquals(1, $element->getAttribute('sequenceIndex'));
+
+        $this->assertEquals(1,$element->getElementsByTagName('responseVariable')->length);
+        $this->assertEquals(1,$element->getElementsByTagName('templateVariable')->length);
+    }
+
+    public function testMarshallMinimal()
+    {
+        $component = new ItemResult(
+            new QtiIdentifier('fixture-identifier'),
+            new DateTime('2018-06-27T09:41:45.529'),
+            1
+        );
+
+        /** @var DOMElement $element */
+        $element = $this->getMarshallerFactory()->createMarshaller($component)->marshall($component);
+
+        $this->assertInstanceOf(DOMElement::class, $element);
+
+        $this->assertEquals($component->getQtiClassName(), $element->nodeName);
+
+        $attributes = [];
+        for ($i=0; $i<2; $i++) {
+            $attributes[] = $element->attributes->item($i)->name;
```

Comment:
can we add a phpdoc tot he element because i am not able to find the name attribute